src/features/commentsSlice.ts

```diff
@@ -0,0 +1,102 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Comment, NewCommentData } from '../types/Comment';
+import {
+  createComment,
+  deleteComment,
+  getPostComments,
+} from '../api/comments';
+
+type CommentsState = {
+  comments: Comment[];
+  loaded: boolean;
+  hasError: boolean;
+  visible: boolean;
+  error: string;
+};
+
+const initialState: CommentsState = {
+  comments: [],
+  loaded: true,
+  hasError: false,
+  visible: false,
+  error: '',
+};
+
+export const init = createAsyncThunk('comments/fetch', (postId: number) => {
+  return getPostComments(postId);
+});
+
+export const addComment = createAsyncThunk(
+  'commentCreate/fetch',
+  (newComment: NewCommentData) => {
+    return createComment(newComment);
+  },
+);
+
+export const deleteCom = createAsyncThunk(
+  'deleteComment/fetch',
```

Comment:
The asyncThunk name should follow the convention 'domain/event'. Here it should be 'comments/deleteComment'.